### DISCUSSION TITLE
HA Z-Wave trigger, Message malformed: Value x-32-0-currentValue can't be found on node Node(node_id=x)

### DISCUSSION THREAD

**Discussion by nmpu** [zwave-js/zwave-js#6575] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

I'm trying to use x-32-0-currentValue to trigger a HA automation based on double-tap. The current trigger device is a Z-Wave 14318/ZW4005. I've associated group 3 (double-tap) with the hub. I can always see the events coming through on debug. Everything works as expected, but you have to prime the pump. The HA trigger always fails to initialize on startup.

```
Error setting up trigger Tap 2x On/Off
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/device_automation/trigger.py", line 64, in async_attach_trigger
    return await platform.async_attach_trigger(hass, config, action, trigger_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/zwave_js/device_trigger.py", line 453, in async_attach_trigger
    return await trigger.async_attach_trigger(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/zwave_js/trigger.py", line 45, in async_attach_trigger
    return await platform.async_attach_trigger(hass, config, action, trigger_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/zwave_js/triggers/value_updated.py", line 205, in async_attach_trigger
    _create_zwave_listeners()
  File "/usr/src/homeassistant/homeassistant/components/zwave_js/triggers/value_updated.py", line 187, in _create_zwave_listeners
    value = node.values[value_id]
            ~~~~~~~~~~~^^^^^^^^^^
KeyError: '34-32-0-currentValue'
```

Once I go **double-tap the switch** and then **disable and enable the automation**, it works. This sure seems like an initialization issue. Maybe I'm just lucky this works at all. Is there a better way to catch the double-tap?

```
alias: Tap 2x On/Off
trigger:
  - platform: device
    device_id: 99a2268aeb554c812f8d42141688b566
    command_class: 32
    endpoint: 0
    property: currentValue
    domain: zwave_js
    type: zwave_js.value_updated.value
```

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

[a comment by an automated account is not shown]

**Comment by nmpu** [comment]
I also have 14294/ZW3005 and 39455/ZW4005 which use the **event** property. These do not fail on startup. Is there some underlying architectural difference? Is it possible that historical feature requests have followed rabbit trails to different camps? Since the **currentValue** method doesn't work now and likely never has, I'd vote for grouping all similar devices under the **event** umbrella. **currentValue** also tells you whether the _previous_ double-tap was up or down. Who cares?

On a side note. you'll see that I've placed an 'x' under **subtype**. It's a _required_ parameter with no apparent affect?

```
  - platform: device
    device_id: 66501ed7f06c278a9779f68dfc87a2dd
    command_class: 32
    endpoint: 0
    property: event
    domain: zwave_js
    type: event.value_notification.basic
    subtype: x
```

**Comment by kpine** [comment]
Events are supported for GE double taps by setting the `treatBasicSetAsEvent` compat flag in the device config file (see [examples](https://github.com/search?q=repo%3Azwave-js%2Fnode-zwave-js+treatBasicSetAsEvent+language%3AJSON&type=code)).

The device can also be auto-enrolled into Group 3 during interview by manually setting the `isLifeline` for group 3, also in the device config file (see [examples](https://github.com/search?q=repo%3Azwave-js%2Fnode-zwave-js+%22isLifeline%22%3A+true+%22Double+tap%22+language%3AJSON&type=code&l=JSON)). 

Be aware that this project is currently in a temporary maintenance mode, so your issue will probably not be addressed immediately: https://github.com/zwave-js/node-zwave-js/issues/6521

If you're using Z-Wave JS UI, you can modify the device config file and add it to the store config directory: https://zwave-js.github.io/zwave-js-ui/#/device-support/custom-device-files, which will let you test the implementation locally.

Details about device config files are at https://zwave-js.github.io/node-zwave-js/#/config-files/overview. If you get it working, consider [contributing](https://zwave-js.github.io/node-zwave-js/#/config-files/contributing-files) the updated config file with a PR.

**Reply by nmpu** [reply]
I made both the suggested changes and now the 14318 behaves like the 14294 and 39455.
